src/check_log.c

```diff
@@ -63,6 +64,27 @@ const char *srunner_log_fname(SRunner * sr)
     return getenv("CK_LOG_FILE_NAME");
 }
 
+enum xml_format srunner_xml_format(SRunner * sr)
+{
+    // if the format as been explicitly set already via
+    // `srunner_set_xml_format`, then use that value
+    if (sr->xml_format != CK_XML_FORMAT_UNSPECIFIED) 
+        return sr->xml_format;
+
+    // junit is the only value of CK_XML_FORMAT_NAME that will
+    // return something other than CK_XML_FORMAT_DEFAULT
+    const char *format_name = getenv("CK_XML_FORMAT_NAME");
```

Comment:
Since it's logging, I made the precedence the same as `srunner_xml_fname`.  This seems sane to me even if it's different than other items. These two should at least agree probably.